llama_index/chat_engine/context.py

```diff
@@ -75,12 +79,22 @@ def from_defaults(
             ]
 
         prefix_messages = prefix_messages or []
-
-        return cls(retriever, llm=llm, memory=memory, prefix_messages=prefix_messages)
+        node_postprocessors = node_postprocessors or []
+
+        return cls(
+            retriever,
+            llm=llm,
+            memory=memory,
+            prefix_messages=prefix_messages,
+            node_postprocessors=node_postprocessors,
+        )
 
     def _generate_context(self, message: str) -> Tuple[str, List[NodeWithScore]]:
         """Generate context information from a message."""
         nodes = self._retriever.retrieve(message)
+        for postprocessor in self._node_postprocessors:
+            nodes = postprocessor.postprocess_nodes(nodes)
```

Comment:
Actually, we should also pass in the message too (have to use the query bundle though, tech debt for another day maybe lol)

Something like:
```
from llama_index.indices.query.schema import QueryBundle
...
postprocess_nodes(nodes, query_bundle=QueryBundle(message))
```

Reply:
TBH I only still the code from [here](https://github.com/jerryjliu/llama_index/blob/main/llama_index/query_engine/citation_query_engine.py#L205) and [here](https://github.com/jerryjliu/llama_index/blob/main/llama_index/evaluation/dataset_generation.py#L99) lol.

So yeah maybe for the next PR, because as per my understanding ideally we also need to pass the embedding to `QueryBundle()`

Reply:
embedding is optional :)

Reply:
I had it changed locally, but couldn't push to this PR branch sadly

Reply:
Could have sworn we fixed those other two spots, whoops

Reply:
All right, fixed.

Hopefully, as I am AFK already, so only able to edit from my phone. 😅